be/src/main/java/yeonba/be/arrow/repository/ArrowTransactionRepositoryImpl.java

```diff
@@ -0,0 +1,32 @@
+package yeonba.be.arrow.repository;
+
+import static yeonba.be.arrow.entity.QArrowTransaction.arrowTransaction;
+import static yeonba.be.user.entity.QUser.user;
+
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.time.LocalDateTime;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class ArrowTransactionRepositoryImpl implements ArrowTransactionRepositoryCustom {
+
+    private final int MAX_ARROW_COUNT_OF_AD = 3;
+    private final JPAQueryFactory queryFactory;
+
+    @Override
+    public boolean canChargeByAdvertisement(long userId, LocalDateTime today) {
+
+        long countOfTodayAdView = queryFactory
+            .select(arrowTransaction)
+            .from(arrowTransaction)
+            .innerJoin(arrowTransaction.receivedUser, user)
+            .where(
+                arrowTransaction.sentUser.isNull(),
+                arrowTransaction.receivedUser.id.eq(userId),
+                arrowTransaction.createdAt.after(today)
+            )
+            .fetchCount();
```

Comment:
```suggestion
        long countOfTodayAdView = Optional.ofNullable(
          queryFactory
              .select(arrowTransaction.count())
              .from(arrowTransaction)
              .innerJoin(arrowTransaction.receivedUser, user)
              .where(
                  arrowTransaction.sentUser.isNull(),
                  arrowTransaction.receivedUser.id.eq(userId),
                  arrowTransaction.createdAt.after(today)
              )
              .fetchOne())
              .orElse(0L);
```
제가 알고 있기론 `fetchCount()`가 deprecated 되어서 count와 `fetchOne()`을 활용하여 카운팅쿼리를 구성하는 방식을 권장하는 것으로 아는데 한 번 살펴보고 수정 부탁드립니다! 제시한 예시에서 `Optional`을 쓴 이유는 `fetchOne()`의 결과로 `null`이 반환될 수 있기 때문입니다.

Reply:
넵 확인했습니다. 수정하겠습니다.